drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2077 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file Driver for PCA(L)xxxx SERIES I2C-based GPIO expander.
+ */
+
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/i2c.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_pca_series);
+
+/* Feature flags */
+#define PCA_HAS_LATCH		BIT(0)	/** + output_drive_strength, + input_latch */
+#define PCA_HAS_PULL		BIT(1)	/** + pull_enable, + pull_select */
+#define PCA_HAS_INT_MASK	BIT(2)	/** + interrupt_mask, + int_status */
+#define PCA_HAS_INT_EXTEND	BIT(3)	/** + interrupt_edge, + interrupt_clear */
+#define PCA_HAS_OUT_CONFIG	BIT(4)	/** + input_status, + output_config */
+
+/* get port and pin from gpio_pin_t */
+#define PCA_PORT(gpio_pin)		(gpio_pin >> 3)
+#define PCA_PIN(gpio_pin)		(gpio_pin & GENMASK(2, 0))
+
+#define PCA_REG_INVALID (0xff)
+
+/**
+ * @brief part number definition.
+ *
+ * @note must be consistent with @ref part_no in dt binding
+ */
+enum gpio_pca_series_part_no {
+	gpio_pca_series_pca9538,
+	gpio_pca_series_pca9539,
+	gpio_pca_series_pca9574,
+	gpio_pca_series_pca9575,
+	gpio_pca_series_pcal9538,
+	gpio_pca_series_pcal9539,
+	gpio_pca_series_pcal6408a,
+	gpio_pca_series_pcal6416a,
+	gpio_pca_series_pcal6524,
+	gpio_pca_series_pcal6534,
+};
+
+/**
+ * @brief part name definition for debug.
+ *
+ * @note must be consistent with @ref part_no in dt binding
+ */
+const char *const gpio_pca_series_part_name[] = {
+	"pca9538",
+	"pca9539",
+	"pca9574",
+	"pca9575",
+	"pcal9538",
+	"pcal9539",
+	"pcal6408a",
+	"pcal6416a",
+	"pcal6524",
+	"pcal6534",
+};
+
+/**
+ * Device reg layout types:
+ * - Type 1: PCA953X, PCA955X
+ * - Type 2: PCAL953X, PCAL955X, PCAL64XX
+ * - Type 3: PCA957X
+ * - Type 4: PCAL6524, PCAL6534
+ */
+
+enum gpio_pca_series_reg_type {				/** Type1 Type2 Type3 Type4 */
```

Comment:
Convention in Zephyr is typically to capitalize all `enum` constants- this comment applies to the additional enums below as well

Reply:
renamed.